### ISSUE TITLE
Indentation rules should be made clearer in list comprehsions

### ISSUE THREAD

**Issue by forki** [dotnet/fsharp#4234]
For elmish newcomers it's sometimes hard to understand the indentation rules. And compiler error messages are not helpful:

    let div _ _ = 1

    let subView _ _ = [1; 2]

    let view1 model dispatch =
       [    
            div [] [
               subView model dispatch
            ]
       ]

    let view2 model dispatch =
       [    
            div [] 
            [
               subView model dispatch
            ]
       ]

    let view3 model dispatch =
       [   
            div [] 
                [
                   subView model dispatch
                ]
       ]

view1 and view3 are ok. view2 gives:

![image](https://user-images.githubusercontent.com/57396/35192899-b0434f3a-fe9a-11e7-9863-fef7d0c00331.png)

**Comment by isaacabraham** [comment]
Did you mean to add an incorrect indent in view 2 as per the screenshot? in the code sample it looks ok

**Comment by forki** [comment]
No, view2 doesn't compile. Try it

**Comment by isaacabraham** [comment]
Ok. I just meant the screenshot you attached shows code that's different from the code snippet

**Comment by forki** [comment]
Yes that closing bracket was indented one char to the left - which is fully
ok.

Am 21.01.2018 14:01 schrieb "Isaac Abraham" <notifications@github.com>:

**Comment by FunctionalFirst** [comment]
How do you propose improving this? To me, the indenting in `view2` indicates the list is at the same level as `div`, not under `div`, as it should be. So the compiler error seems correct.

**Comment by cartermp** [comment]
The compiler error is correct, but it’s not easy to trace back to the root of the issue, which is failing to indent the list passed as a parameter to the div function.

**Comment by isaacabraham** [comment]
As with all of these errors, they're always correct, but unfortunately next to useless to actually guide the developer to the solution ;)

**Comment by forki** [comment]
see #4255
